### PR TITLE
document limits on using dots in service names

### LINKED ISSUE #1205: Can't grab instances that service.Name include "."
Consul include two services：

> consul
> go.micro.api.greeter

### Configuration
Consul Template configuration file(app.ctmpl):

```
{{range services}}
 	 upstream {{.Name}} {
    	 {{range service .Name}}
   	     server {{.Address}}:{{.Port}};
	 {{end}}
    }
{{end}}
```

### Shell

> consul-template -template "app.ctmpl:nginx.conf" -once

### nginx.conf
```
upstream consul { 
   	     server 127.0.0.1:8300; 
    }   
 upstream go.micro.api.greeter {	 
   }
```

why instances of "go.micro.api.greeter" is not be found?

### PR DESCRIPTION
**Pull request by eikenb**
Fixes #1205

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -981,6 +981,9 @@ This will returns services which are deemed "passing" or "warning" according to
 their node and service-level checks defined in Consul. Please note that the
 comma implies an "or", not an "and".
 
+**Note:** Due to the use of dot `.` to delimit TAG, the `service` command will
+not recognize service names containing docs.
```
Comment (judithpatudith) [inline README.md]:
```suggestion
not recognize service names containing dots.
```

File: README.md [hunk 2]
```diff
@@ -1862,6 +1866,14 @@ func main() {
 
 ## Caveats
 
+### Dots in Service Names
+
+Using dots `.` in service names will conflict with the use of dots for [TAG
+delineation](https://github.com/hashicorp/consul-template#service) in the
+template. Dots already [interfere with using
+DNS](https://www.consul.io/docs/agent/services.html#service-and-tag-names-with-dns)
+for service names, so it is really just best to not use dots if possible.
```
Comment (judithpatudith) [inline README.md]:
```suggestion
for service names, so we recommend avoiding dots wherever possible.
```

File: README.md (hunk 1 above)
Comment (eikenb) [reply README.md]: Fixed. Thanks.

File: README.md (hunk 2 above)
Comment (eikenb) [reply README.md]: That sounds much better. Updated. Thanks.

**Pull request merged**